public/pages/Findings/components/FindingDetailsFlyout.tsx

```diff
@@ -319,46 +326,60 @@ export default class FindingDetailsFlyout extends Component<
     return patternId;
   };
 
-  toggleDocumentDetails(item: FindingDocumentItem) {
-    const docIdToExpandedRowMapValues = { ...this.state.docIdToExpandedRowMap };
+  createDocumentBlock = (item: FindingDocumentItem) => {
     let formattedDocument = '';
     try {
       formattedDocument = document ? JSON.stringify(JSON.parse(item.document), null, 2) : '';
     } catch {
       // no-op
     }
 
+    return (
+      <EuiFormRow fullWidth={true} style={{ width: '100%' }}>
+        <EuiCodeBlock
+          language="json"
+          isCopyable
+          data-test-subj={`finding-details-flyout-rule-document-${item.itemIdx}`}
+        >
+          {formattedDocument}
+        </EuiCodeBlock>
+      </EuiFormRow>
+    );
+  };
+
+  getRelatedDocuments() {
+    const {
+      finding: { document_list, related_doc_ids },
```

Comment:
do we need to handle cases where finding or its properties are null or undefined?

Reply:
That's a good point, let me think about this. I would rather do this validation at the fetch layer (i.e. in the service or store) so that we don't have to validate findings in all components using them

Reply:
We should also have tests in place in the backend to verify that the API response matches the documented contract